src/CTA.WebForms2Blazor/FileConverters/FileConverter.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+using System.Collections.Generic;
+using CTA.WebForms2Blazor.FileInformationModel;
+
+namespace CTA.WebForms2Blazor.FileConverters
+{
+    public abstract class FileConverter
+    {
+        private readonly string _relativePath;
+
+        public string RelativePath { get { return _relativePath; } }
+
+        protected FileConverter(string relativePath)
+        {
+            _relativePath = relativePath;
+        }
+
+        public abstract Task<IEnumerable<FileInformation>> MigrateFileAsync();
+
+        protected abstract byte[] GetFileBytes();
```

Comment:
Have you and @comsemer discussed if getting the file bytes will differ based on the type of file being processed? My thought is that it would be identical for all files in which case `GetFileBytes()` can become a public method with an implementation. This would save work on each child class of `FileConverter`

Reply:
I thought some of the other types of files we might do some processing before converting to byte[], but I can discuss again to confirm

Reply:
We can also provide a base implementation and only override it if needed for exceptions. Another approach would be to store a base version of the implementation and reference that in all common method and use different code for the exceptions.